server-spi/src/main/java/org/keycloak/organization/OrganizationProvider.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.organization;
+
+import java.util.stream.Stream;
+import org.keycloak.models.ModelDuplicateException;
+import org.keycloak.models.OrganizationModel;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.provider.Provider;
+
+public interface OrganizationProvider extends Provider {
+
+    /**
+     * Creates a new organization with given {@code name} to the given realm.
+     * The internal ID of the organization will be created automatically.
+     * @param realm Realm owning this organization.
+     * @param name String name of the organization.
+     * @throws ModelDuplicateException If there is already an organization with the given name 
+     * @return Model of the created organization.
+     */
+    default OrganizationModel createOrganization(RealmModel realm, String name) {
```

Comment:
We don't need to pass over a `realm` if they are already available from the `KeycloakSession`. It should make the contract simper.

The same goes for other methods.

Reply:
I know, but it just complicates and should leverage more the context from the Keycloak session. Other providers also do the same where the realm is obtained from the session.

Reply:
I probably do not fully understand the complication you mentioned. It seems to me you would be able to use the realm from the session upon calling the provider. And the realm might be then accessible in storage layer where you should not have an access to the session.